pandas/core/frame.py

```diff
@@ -1205,13 +1205,24 @@ def to_records(self, index=True, convert_datetime64=True):
         index : boolean, default True
             Include index in resulting record array, stored in 'index' field
         convert_datetime64 : boolean, default True
+            .. deprecated:: 0.23.0
+
             Whether to convert the index to datetime.datetime if it is a
             DatetimeIndex
 
         Returns
         -------
         y : recarray
         """
+
+        if convert_datetime64 is not None:
```

Comment:
@jorisvandenbossche how should we handle this here. We really want to change the default to False in the future, but need True to make this backwards compat for now.

Reply:
on 2nd thought, let's just change this to default to ``False``. We have to change it now or later anyways.

Reply:
I think we need to have to `None` as a default to detect the case that people now did `convert_datetime64=False` explicitly. Otherwise they don't see the deprecation warning, and their code will break once we remove the keyword.